grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2117,13 +2120,14 @@ class StockMovementService {
             picklistItem.disableRefresh = Boolean.TRUE
             picklistItem.picklist?.removeFromPicklistItems(picklistItem)
             picklistItem.requisitionItem?.removeFromPicklistItems(picklistItem)
+            binLocations.add(picklistItems?.binLocation?.id)
```

Comment:
Should this be `picklistItem?.binLocation?.id`

Reply:
This was causing the following stacktrace
```
2023-05-11 15:03:36,185 [http-8080-4] INFO  filters.AccessLogFilters  - stockMovementApi.reviseItems: [user:jmiranda, location:Distribution Center]
2023-05-11 15:03:36,193 [http-8080-4] INFO  filters.AccessLogFilters  - No rule for stockMovementApi:reviseItems -> allow anonymous
2023-05-11 15:03:36,198 [http-8080-4] INFO  core.UserService  - Is role ROLE_FINANCE in [ROLE_MANAGER, ROLE_AUTHENTICATED, ROLE_BROWSER, ROLE_ADMIN, ROLE_SUPERUSER, ROLE_ASSISTANT] = false
2023-05-11 15:03:36,198 [http-8080-4] INFO  core.UserService  - Is role ROLE_SUPERUSER in [ROLE_MANAGER, ROLE_AUTHENTICATED, ROLE_BROWSER, ROLE_ADMIN, ROLE_SUPERUSER, ROLE_ASSISTANT] = true
2023-05-11 15:03:36,266 [http-8080-4] INFO  inventory.StockMovementService  - Shipment workflow Stock movement shipment workflow
2023-05-11 15:03:36,385 [http-8080-4] INFO  api.StockMovementApiController  - binding lineItems: [[id:ff808181880c4df001880c53118d0003, reasonCode:LOW_STOCK, quantityRevised:1]]
2023-05-11 15:03:36,399 [http-8080-4] INFO  inventory.StockMovementService  - Removing previous changes, picklists and shipments, if present
2023-05-11 15:03:36,527 [http-8080-4] INFO  inventory.ProductAvailabilityService  - refreshProductsAvailability: 8a8a9e9665c4f59d0165c54ec6b10027, [10001], [[8a8a9e96687c94ce0168bab1c5a8285c]], false
2023-05-11 15:03:36,532 [pool-2-thread-1] INFO  inventory.RefreshProductAvailabilityEventService  - Application event been disabled by event publisher
2023-05-11 15:03:36,532 [pool-2-thread-1] INFO  inventory.RefreshProductAvailabilityEventService  - Application event [disableRefresh:true, forceRefresh:false, class:class org.pih.warehouse.inventory.RefreshProductAvailabilityEvent, locationId:8a8a9e9665c4f59d0165c54ec6b10027, productIds:[10001], synchronousRequired:false, timestamp:1683835416503, metaClass:org.codehaus.groovy.runtime.HandleMetaClass@4ba20442[groovy.lang.ExpandoMetaClass@4ba20442[class org.pih.warehouse.inventory.RefreshProductAvailabilityEvent]], source:org.pih.warehouse.picklist.PicklistItem : ff808181880c4df001880c59fde2000d]
2023-05-11 15:03:36,549 [http-8080-4] ERROR errors.GrailsExceptionResolver  - Exception occurred when processing request: [POST] /openboxes/api/stockMovements/ff808181880c4df001880c528d670001/reviseItems
Stacktrace follows:
org.hibernate.TypeMismatchException: Provided id of the wrong type for class org.pih.warehouse.core.Location. Expected: class java.lang.String, got class java.util.ArrayList
	at org.pih.warehouse.inventory.ProductAvailabilityService$_refreshProductsAvailability_closure3.doCall(ProductAvailabilityService.groovy:106)
	at org.pih.warehouse.inventory.ProductAvailabilityService.refreshProductsAvailability(ProductAvailabilityService.groovy:104)
	at org.pih.warehouse.inventory.ProductAvailabilityService$$FastClassByCGLIB$$7ee8fd31.invoke(<generated>)
	at net.sf.cglib.proxy.MethodProxy.invoke(MethodProxy.java:149)
	at org.pih.warehouse.inventory.ProductAvailabilityService$$EnhancerByCGLIB$$80269ed0.refreshProductsAvailability(<generated>)
	at org.pih.warehouse.inventory.ProductAvailabilityService$refreshProductsAvailability.call(Unknown Source)
	at org.pih.warehouse.inventory.StockMovementService.removeShipmentAndPicklistItemsForModifiedRequisitionItem(StockMovementService.groovy:2130)
	at org.pih.warehouse.inventory.StockMovementService$_reviseItems_closure65.doCall(StockMovementService.groovy:1944)
	at org.pih.warehouse.inventory.StockMovementService.reviseItems(StockMovementService.groovy:1934)
	at org.pih.warehouse.inventory.StockMovementService$$FastClassByCGLIB$$b65c0b56.invoke(<generated>)
	at net.sf.cglib.proxy.MethodProxy.invoke(MethodProxy.java:149)
	at org.pih.warehouse.inventory.StockMovementService$$EnhancerByCGLIB$$456057a7.reviseItems(<generated>)
	at org.pih.warehouse.inventory.StockMovementService$reviseItems.call(Unknown Source)
	at org.pih.warehouse.api.StockMovementApiController$_closure12.doCall(StockMovementApiController.groovy:265)
	at org.pih.warehouse.api.StockMovementApiController$_closure12.doCall(StockMovementApiController.groovy)
	at grails.plugin.springcache.web.GrailsFragmentCachingFilter.doFilter(GrailsFragmentCachingFilter.groovy:66)
	at net.sf.ehcache.constructs.web.filter.Filter.doFilter(Filter.java:86)
	at org.grails.plugin.resource.DevModeSanityFilter.doFilter(DevModeSanityFilter.groovy:44)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
@jmiranda yes this should be `picklistItem?.binLocation?.id` (singular item)